### PR TITLE
[RSDK-7316]   Fail CI of erh/viamrtsp if any go unit tests fail

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-7316](https://viam.atlassian.net/browse/RSDK-7316)

- [x] Add phony make target `test` to call recursive run of go test
- [x] Add workflow step to call said target
- [x] Rename workflow yml file to reflect change 
- [x] Test workflow https://github.com/erh/viamrtsp/actions/runs/9271772883/job/25507979569?pr=30

### CODE REVIEW

**Review by randhid — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -44,6 +44,9 @@ lint: gofmt tool-install
 	export pkgs="`go list -f '{{.Dir}}' ./...`" && echo "$$pkgs" | xargs go vet -vettool=$(TOOL_BIN)/combined
 	GOGC=50 $(TOOL_BIN)/golangci-lint run -v --fix --config=./etc/.golangci.yaml
 
+test:
+	go test ./...
```
Comment (nicksanford) [inline Makefile:48]:
```suggestion
	go test -race -v ./...
```

File: .github/workflows/quality-checks.yml [hunk 2]
```diff
@@ -39,3 +39,6 @@ jobs:
 
     - name: Run linter and static checks
       run: make lint
+
```
Comment (nicksanford) [inline .github/workflows/quality-checks.yml:42]: What platforms do we run this on? If we only run it on one platform can you please create follow up tickets to add test runners for at least linux/amd64 and linux/arm64?

File: Makefile (hunk 1 above)
Comment (nicksanford) [reply Makefile:48]: lets add race detection and log verbosely so we know what tests are running

File: .github/workflows/quality-checks.yml (hunk 2 above)
Comment (hexbabe) [reply .github/workflows/quality-checks.yml:42]: Just linux/amd64

[Done](https://viam.atlassian.net/browse/RSDK-7766?atlOrigin=eyJpIjoiNjllZmQ5NTExMWVjNDgwMDk2YWQ3ZmExNzMzYzFkODEiLCJwIjoiaiJ9)

**Review by nicksanford — APPROVED** [review]

**Comment by hexbabe** [comment]
Thanks for the reviews! Gonna merge

**Pull request merged**